cmd/earthly/main.go

```diff
@@ -2194,13 +2242,18 @@ func (app *earthlyApp) actionBuild(c *cli.Context) error {
 			return errors.New("cannot use --no-output with image or artifact modes")
 		}
 	}
-	var target domain.Target
-	var artifact domain.Artifact
-	destPath := "./"
+
 	flagArgs, nonFlagArgs, err := variables.ParseFlagArgsWithNonFlags(c.Args().Slice())
 	if err != nil {
 		return errors.Wrapf(err, "parse args %s", strings.Join(c.Args().Slice(), " "))
 	}
+
+	return app.actionBuildImp(c, flagArgs, nonFlagArgs)
+}
+func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []string) error {
+	var target domain.Target
+	var artifact domain.Artifact
+	destPath := "./"
```

Comment:
I think it's possible to run `earthly --artifact docker2earthly` and cause artifact mode to be enabled, which wouldn't make sense in this case.

Not sure what's the right way to prevent this. In any case - that can be a separate PR.